src/chains/ethereum/block/src/runtime-block.ts

```diff
@@ -156,21 +164,30 @@ export class RuntimeBlock {
       header.mixHash,
       BUFFER_8_ZERO // nonce
     ];
-    if (this.serializeBaseFeePerGas && header.baseFeePerGas !== undefined) {
-      rawHeader[15] = Quantity.toBuffer(header.baseFeePerGas);
+    const isEip4895 = this._common.isActivatedEIP(4895);
+    // baseFeePerGas was added in London
+    if (header.baseFeePerGas !== undefined) {
+      rawHeader[15] = Quantity.toBuffer(header.baseFeePerGas, false);
+      // withdrawalsRoot was added in Shanghai
+      if (isEip4895) rawHeader[16] = Data.toBuffer(header.withdrawalsRoot);
     }
 
     const { totalDifficulty } = header;
-    const txs: TypedDatabaseTransaction[] = [];
-    const extraTxs: GanacheRawBlockTransactionMetaData[] = [];
-    transactions.forEach(tx => {
-      txs.push(<TypedDatabaseTransaction>tx.raw);
-      extraTxs.push([tx.from.toBuffer(), tx.hash.toBuffer()]);
-    });
-    const { serialized, size } = serialize([
-      rawHeader,
-      txs,
-      [],
+    const txs: BlockRawTransaction[] = Array(transactions.length);
+    const extraTxs: GanacheRawBlockTransactionMetaData[] = Array(
+      transactions.length
+    );
+    for (let i = 0; i < transactions.length; i++) {
+      const tx = transactions[i];
+      txs[i] =
+        tx.raw.length === 9
+          ? tx.raw
```

Comment:
I like a comment, although I'm not sure this is the correct one?
```suggestion
          ? tx.raw // legacy transactions are correctly encoded
```

Reply:
```suggestion
          ? tx.raw // legacy transactions don't have their own encoding
```